datafusion/sqllogictest/test_files/sort_merge_join.slt

```diff
@@ -84,7 +84,6 @@ SELECT * FROM t1 LEFT JOIN t2 ON t1.a = t2.a AND t2.b * 50 <= t1.b
 Alice 100 Alice 1
 Alice 100 Alice 2
 Alice 50 Alice 1
-Alice 50 NULL NULL
```

Comment:
does it mean tests were not correct?

Reply:
Yes, the results were incorrect.

Reply:
For example, as `Alice 50` is joined with `Alice 1`, it should not be joined with nulls.

Reply:
Agree, just checked in DuckDB
```
D CREATE TABLE t1(a text, b int); insert into t1  VALUES ('Alice', 50), ('Alice', 100), ('Bob', 1);
D CREATE TABLE t2(a text, b int);
D insert into t2 VALUES ('Alice', 2), ('Alice', 1);
D SELECT * FROM t1 LEFT JOIN t2 ON t1.a = t2.a AND t2.b * 50 <= t1.b
>
> ;
┌─────────┬───────┬─────────┬───────┐
│    a    │   b   │    a    │   b   │
│ varchar │ int32 │ varchar │ int32 │
├─────────┼───────┼─────────┼───────┤
│ Alice   │    50 │ Alice   │     1 │
│ Alice   │   100 │ Alice   │     1 │
│ Alice   │   100 │ Alice   │     2 │
│ Bob     │     1 │         │       │
└─────────┴───────┴─────────┴───────┘
```